src/sw.js

```diff
@@ -0,0 +1,39 @@
+var window = {};
+
+self.addEventListener('install', function (event) {
+  // Put `offline.html` page into cache
+  var offlineRequest = new Request('offline.html');
+  event.waitUntil(
+    fetch(offlineRequest).then(function (response) {
+      return caches.open('offline').then(function (cache) {
+        console.log('[oninstall] Cached offline page', response.url);
+        return cache.put(offlineRequest, response);
+      });
+    })
+  );
+});
+
+self.addEventListener('fetch', function (event) {
+  // Only fall back for HTML documents.
+  var request = event.request;
+  // && request.headers.get('accept').includes('text/html')
+  if (request.method === 'GET') {
+    // `fetch()` will use the cache when possible, to this examples
+    // depends on cache-busting URL parameter to avoid the cache.
+    event.respondWith(
+      fetch(request).catch(function (error) {
+        // `fetch()` throws an exception when the server is unreachable but not
```

Comment:
Ok Just removed uncoment code or write the description, why we have to have this comment.